# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: intl/locale/LocaleService.cpp
# Commit: aca7cd850cc9
# Full Hash: aca7cd850cc9c3a88dfcd82d58268929aa288497
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-07-08 09:42:17
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1627075 - Route Omnijar requests through StartupCache r=froydnj
#   
#   This should be a relatively straightforward patch. Essentially, we implement
#   a wrapper class (and friends) around nsZipArchive (and friends), which transparently
#   caches entries from the underlying zip archive in the StartupCache. This will break
# ==============================================================================

diff -r d2e40366118c -r aca7cd850cc9 intl/locale/LocaleService.cpp
--- a/intl/locale/LocaleService.cpp	Wed Jul 08 02:46:11 2020 +0000
+++ b/intl/locale/LocaleService.cpp	Wed Jul 08 02:46:34 2020 +0000
@@ -295,13 +295,14 @@
 
 static bool GetGREFileContents(const char* aFilePath, nsCString* aOutString) {
   // Look for the requested file in omnijar.
-  RefPtr<nsZipArchive> zip = Omnijar::GetReader(Omnijar::GRE);
+  RefPtr<CacheAwareZipReader> zip = Omnijar::GetReader(Omnijar::GRE);
   if (zip) {
-    nsZipItemPtr<char> item(zip, aFilePath);
-    if (!item) {
+    uint32_t length;
+    const uint8_t* data = zip->GetData(aFilePath, &length);
+    if (!data) {
       return false;
     }
-    aOutString->Assign(item.Buffer(), item.Length());
+    aOutString->Assign(reinterpret_cast<const char*>(data), length);
     return true;
   }
 